src/toast/tod/sss.py

```diff
@@ -0,0 +1,242 @@
+# Copyright (c) 2015-2019 by the parties listed in the AUTHORS file.
+# All rights reserved.  Use of this source code is governed by
+# a BSD-style license that can be found in the LICENSE file.
+
+import os
+
+import numpy as np
+
+from ..utils import Logger
+
+from ..timing import function_timer, Timer
+from ..rng import random
+
+from ..op import Operator
+
+from toast.mpi import MPI
+
+import toast.qarray as qa
+
+import healpy as hp
+
+
+class OpSimScanSynchronousSignal(Operator):
+    """Operator which generates scan-synchronous signal timestreams.
+    
+    Args:
+        out (str): accumulate data to the cache with name
+            <out>_<detector>.  If the named cache objects do not exist,
+            then they are created.
+        realization (int): if simulating multiple realizations, the
+            realization index.
+        component (int): the component index to use for this noise
+            simulation.
+        nside (int): ground map healpix resolution
+        fwhm (int): ground map smoothing scale
+        lmax (int): ground map expansion order
+        scale (float): RMS of the ground signal fluctuations at el=45deg
+        power (float): exponential for suppressing ground pickup at
+             higher observing elevation
+        report_timing (bool):  Print out time taken to initialize,
+             simulate and observe
+    """
+
+    def __init__(
+        self,
+        out="sss",
+        realization=0,
+        component=663056,
+        nside=128,
+        fwhm=10,
+        lmax=256,
+        scale=1e-3,
+        power=-1,
+        report_timing=False,
+    ):
+        # Call the parent class constructor
+        super().__init__()
+
+        self._out = out
+        self._realization = realization
+        self._component = component
+        self._nside = nside
+        self._lmax = lmax
+        self._fwhm = fwhm
+        self._scale = scale
+        self._power = power
+        self._report_timing = report_timing
+        return
+
+    @function_timer
+    def exec(self, data):
+        """Generate timestreams.
+
+        This iterates over all observations and detectors and generates
+        the atmosphere timestreams.
+
+        Args:
+            data (toast.Data): The distributed data.
+
+        Returns:
+            None
+
+        """
+
+        log = Logger.get()
+        group = data.comm.group
+        for obs in data.obs:
+            try:
+                obsname = obs["name"]
+            except Exception:
+                obsname = "observation"
+            prefix = "{} : {} : ".format(group, obsname)
+            tod = self._get_from_obs("tod", obs)
+            comm = tod.mpicomm
+            rank = 0
+            if comm is not None:
+                rank = comm.rank
+            site = self._get_from_obs("site_id", obs)
```

Comment:
flake warning:  "site is assigned to but never used"  (leftover from atm copy / paste).